project/src/main/webapp/gmail.js

```diff
@@ -15,191 +15,235 @@
+            return false;
+        }
+        return true;
     }
 
-
-    async function keywordExtraction(userInput) {
-    var result = "";
-    var count = 0;
-    var searchQuery = userInput.split(" ");
-    for (word of searchQuery){
-        if ((searchQuery.length-1) == count && word.indexOf('?') != -1) {
-                word = word.substring(0, word.length-1);
-            }
-        var url = new URL("https://api.datamuse.com/words?md=p&sp=" + word + "&max=1");
-        const output = await fetch(url).then(response => response.text());
-        if (output.includes("[\"n\"]") == true || output.includes("[\"adj\"]") == true){
-            result += word;
-            if (count != (searchQuery.length-1)){
-                result += " ";
-            }
-        }
-        count++;
+    getEmailObjects() {
+        console.log(this.emailObjects);
     }
-    return result; 
+
+    /**
+     * Get Message with given ID.
+     *
+     * @param  {String} messageId ID of Message to get.
+     */
+    getMessage(messageId) {
+        const getMessagePromise = new Promise((resolve) => {
+            const messageRequest = gapi.client.gmail.users.messages.get({
+                'userId': 'me',
+                'id': messageId,
+                'format': "raw"
+            });
+            messageRequest.execute(response => {
+                //Convert from base64 encoding to text.
+                const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+                const emailBodyValue = this.getEmailBody(decodedEmail);
+                const emailDateValue = this.getEmailDate(decodedEmail);
+                const messageID = this.getMessageID(decodedEmail);
+                if (this.isActualEmail(emailBodyValue, emailDateValue)) {
+                    this.emailObjects[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+                }
+                this.getEmailObjects();
+                resolve();
+            });
+        });
+        return getMessagePromise;
     }
-    
+
     //Retrieve messages using hardcoded queries and the signed-in email.
-    async function listMessages() {
-        const t0 = performance.now();
-        var list = [];
-        var getPageOfMessages = function(request, result) {
-            request.execute(function(resp) {
-                result = result.concat(resp.messages);
-                var nextPageToken = resp.nextPageToken;
-                if (nextPageToken) {
-                    request = gapi.client.gmail.users.messages.list({
-                    'userId': 'me',
-                    'pageToken': nextPageToken,
-                    'q': "When is the Interview Prep Workshop?"
-                });
-                getPageOfMessages(request, result);
-                } else {
-                    for (i = 0; i < result.length; i++) {
-                        const messageRequest = gapi.client.gmail.users.messages.get({
+    listMessages(question) {
```

Comment:
I will approve this function since you have updated it in https://github.com/Enoch-Chou/capstoneproject/pull/12/files#r474320469 and it will be a pain to revert these changes.
Just make sure that #18 gets merged before #12